### ISSUE TITLE
Running as root is not supported at the moment.

### ISSUE THREAD

**Issue by Luckytiger666** [OpenDevin/OpenDevin#887]
The Traceback is like: 
 
![image](https://github.com/OpenDevin/OpenDevin/assets/44977012/bbbeb8e1-f9e0-4e0e-818d-feb6347a0b8d)

Here is my config:
![image](https://github.com/OpenDevin/OpenDevin/assets/44977012/4b2c0fa3-d15d-42ff-96e0-f68b9fd815ae)

**Comment by SmartManoj** [comment]
How did you run?

**Comment by Luckytiger666** [comment]
Just follow the instruction.
1. pull code
2. make install
3. make run

I did not run it as a root user.

by the way, my OS is Mac.

**Comment by li-boxuan** [comment]
I am using Mac (M1) and met the same problem.

**Comment by Luckytiger666** [comment]
![image](https://github.com/OpenDevin/OpenDevin/assets/44977012/5be87d0e-e33b-421e-85ca-59a8d3f39eec)
It seems Docker status is OK.

**Comment by SmartManoj** [comment]
Please follow issue template

#### Setup and configuration
**Current version**:

```bash
```

**Comment by Luckytiger666** [comment]
OK

**Comment by Luckytiger666** [comment]
#### Setup and configuration
**Current version**:
```
commit cc6626ff0da9f3c146e88c179f1e1432b9d60550 (HEAD -> main, origin/main, origin/HEAD)
Author: Robert Brennan <accounts@rbren.io>
Date:   Mon Apr 8 01:39:36 2024 -0500

    fix up json parsing (#875)
```

**My config.toml and environment vars** (be sure to redact API keys):
```
LLM_API_KEY="ollama"
LLM_MODEL="ollama/llama2:13b-chat-q4_K_M"
LLM_EMBEDDING_MODEL="local"
LLM_BASE_URL="http://localhost:11434"
WORKSPACE_DIR="./workspace"
```
**My model and agent** (you can see these settings in the UI):
* Model: ollama/llama2:13b
* Agent: MonologueAgent

**Commands I ran to install and run OpenDevin**:
```
make build
make run
```

**Steps to Reproduce**:
1.  install dependences.
2. try to run server.
3. got problem above.

**Logs, error messages, and screenshots**:
##### Logs:
```
15:32:55 - opendevin:ERROR: manager.py:134 - Error creating controller.
Traceback (most recent call last):
  File "/Users/zhangbowen/OpenDevin/OpenDevin/opendevin/server/agent/manager.py", line 125, in create_controller
    self.controller = AgentController(
                      ^^^^^^^^^^^^^^^^
  File "/Users/zhangbowen/OpenDevin/OpenDevin/opendevin/controller/agent_controller.py", line 86, in __init__
    self.command_manager = CommandManager(
                           ^^^^^^^^^^^^^^^
  File "/Users/zhangbowen/OpenDevin/OpenDevin/opendevin/controller/command_manager.py", line 14, in __init__
    self.shell = DockerInteractive(
                 ^^^^^^^^^^^^^^^^^^
  File "/Users/zhangbowen/OpenDevin/OpenDevin/opendevin/sandbox/sandbox.py", line 142, in __init__
    raise NotImplementedError(
NotImplementedError: Running as root is not supported at the moment.
```
##### WebPage screen shot:
![image](https://github.com/OpenDevin/OpenDevin/assets/44977012/71cda0c4-cb43-498d-9531-36fdd4d772c5)

**Comment by yehudi-dev** [comment]
I use Mac M3 and have the same issue. Workaround by change `ConfigType.RUN_AS_DEVIN: 'true'` in config.py (but still have other issues for websocket connection)

**Comment by hangatzu2017** [comment]
Same problem here,
I'm using WSL2 on windows 11.

**Comment by li-boxuan** [comment]
Seems to be a regression introduced by https://github.com/OpenDevin/OpenDevin/pull/847/files

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ve-cattozzi** [comment]
Same problem for me,
I'm using WSL2 on windows 11.

**Comment by adriano-kayzen** [comment]
It also happens when trying to run using OpenAI

Config.toml:
``` 
LLM_MODEL="gpt-3.5-turbo-1106"
LLM_API_KEY="SOME_KEY"
LLM_EMBEDDING_MODEL=""
WORKSPACE_DIR="./workspace"
```

**Comment by yufansong** [comment]
It should solved by the hotfix #895, close it now. If new problem happend, please one a new issue.

**Comment by todi-03** [comment]
same problem with my mackbook air m2
